includes/class-woothemes-sensei-quiz.php

```diff
@@ -1053,4 +1056,35 @@ public function get_user_question_feedback( $lesson_id, $question_id, $user_id =
 
      } // end get_user_question_feedback
 
+     /**
+      * Check if a quiz has no questions, and redirect back to lesson.
+      *
+      * Though a quiz is created for each lesson, it should not be visible
+      * unless it has questions.
+      *
+      * @since 1.9.0
+      * @access public
+      * @param none
+      * @return void
+      */
+
+     public function quiz_has_no_questions() {
+
+         global $post;
+
+         $lesson_id = $this->get_lesson_id($post->ID);
+
+         $has_questions = get_post_meta( $lesson_id, '_quiz_has_questions', true );
```

Comment:
@davidpagepsycle Valid point, in that case relying on this meta can be considered a stable option.